flow360/component/simulation/translator/utils.py

```diff
@@ -156,15 +156,42 @@ def _is_last_of_type(lst, obj):
     return lst[last_index] == obj
 
 
+def getattr_recursive(obj, attr_name: (str, list), *args):
+    """Get attribute recursively from a list"""
+    # If attr_name is a string, return the attribute directly
+    if isinstance(attr_name, str):
+        return getattr(obj, attr_name, *args)
+
+    # If attr_name is a list, iterate through each attribute name
+    for attr in attr_name:
+        obj = getattr(obj, attr)
+
+    return obj
+
+
+def get_default_recursive(obj, attr_name: (str, list)):
+    """Get field default recursively from a list"""
+    if isinstance(attr_name, str):
+        return obj.model_fields[attr_name].default
+
+    for attr in attr_name:
+        obj = getattr(obj, attr)
```

Comment:
This won't work getting defaults right?

Reply:
We are getting default in the final level